src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -422,9 +439,9 @@ private void AddLexedTokenSlot()
             }
             else
             {
-                var tmp = new ArrayElement<SyntaxToken>[_lexedTokens.Length * 2];
-                Array.Copy(_lexedTokens, tmp, _lexedTokens.Length);
-                _lexedTokens = tmp;
+                var old = _lexedTokens;
+                Array.Resize(ref _lexedTokens, _lexedTokens.Length * 2);
```

Comment:
I'm not sure why we would not attempt to retrieve from the pool in this scenario, or to discard the old array. Surely the intermediately-sized arrays have value, in that we need to work our way up through them to get to larger arrays? In other words, for one particularly large file, I'd expect us to populate the pool with all the intermediately-sized arrays (up to the max length we track), benefiting all other non-large files of those intermediate sizes.

Reply:
I don't believe that the ObjectPool class being used has an Allocate method that takes in a size hint. So, if we did use the Allocate method to obtain a new array, there is no guarantee that it would be large enough for our needs. I suppose we could not forget the old array, but the value of doing that seems questionable. As an aside, the approach taken here closely mimics what the AddTokenSlot method does directly above.